### PR TITLE
rgb_html() function

### PR DESCRIPTION
**Pull request by geoffco**
* updated example doc to match current I2C initiator
* added rgb_html() function to return HTML hex representation of RGB values

### CODE REVIEW

File: tcs34725.py [hunk 1]
```diff
@@ -123,6 +123,23 @@ def _temperature_and_lux(self, data):
...
+        red = int((r/c) * 256)
+        green = int((g/c) * 256)
+        blue = int((b/c) * 256)
+
+        gammatable = {}
+        for x, y in enumerate(range(0, 256)):
+            x = x / 255
+            x = pow(x, 2.5)
+            x = x * 255
+            gammatable[y] = x
```
Comment (deshipu) [inline tcs34725.py]: It would be much better to compute this on the fly for the three values you are going to use it for, instead of allocating a big chunk of memory and computing this for all the 256 possible values.

File: tcs34725.py [hunk 2]
```diff
@@ -123,6 +123,23 @@ def _temperature_and_lux(self, data):
         cct = 449.0 * n**3 + 3525.0 * n**2 + 6823.3 * n + 5520.33
         return cct, y
 
+    def html_rgb(self):
+        r, g, b, c = self.read(raw=True)
+        red = int((r/c) * 256)
+        green = int((g/c) * 256)
+        blue = int((b/c) * 256)
```
Comment (deshipu) [inline tcs34725.py]: Instead of calling the `self.read` yourself, it would be better to just provide a conversion function that takes a 4-tuple as the parameter and returns the html rgb, or even better, returns a 3-tuple.

Also, documentation is missing.

File: docs/examples.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@ Now, to make basic measurement::
 
     import tcs34725
     from machine import I2C, Pin
-    i2c = I2C(Pin(5), Pin(4))
+    i2c = I2C(-1, Pin(5), Pin(4))
```
Comment (deshipu) [inline docs/examples.rst]: Can you put that in a separate pull request, as it is not related to the feature you are adding?

**Review by deshipu — COMMENTED** [review]
Thank you very much for your contribution. I think it's generally a good idea, but I would prefer a but I think a bit different implementation would make more sense.

**Comment by geoffco** [comment]
Thanks for the feedback. 
I've implemented your suggestions with one modification - returning both the 8 bit RGB values and the hex string to save writing a separate function or having to do the conversion each time.

File: tcs34725.py [hunk 4]
```diff
@@ -123,22 +123,17 @@ def _temperature_and_lux(self, data):
...
+
+        r, g, b, c = data
+        red = pow((int((r/c) * 256) / 255), 2.5) * 255
+        green = pow((int((g/c) * 256) / 255), 2.5) * 255
+        blue = pow((int((b/c) * 256) / 255), 2.5) * 255
+
+        rgbhex = "{0:02x}{1:02x}{2:02x}".format(int(red),
+                                 int(green),
+                                 int(blue))
+        return red, green, blue, rgbhex
```
Comment (deshipu) [inline tcs34725.py]: It doesn't need to be a method anymore. Can you make it a normal function?

Also, I would make the html format a separate function after all, possibly calling the rgb function internally. The reason is, you really ever need one or the other, never both.

**Review by deshipu — COMMENTED** [review]
Thank you, that looks much better! Two more comments.

**Comment by geoffco** [comment]
html_rgb and html_hex are now separate functions outside the class.

**Review by deshipu — APPROVED** [review]

**Comment by deshipu** [comment]
Thank you very much!
